src/ucp/core/ucp_ep.c

```diff
@@ -214,10 +214,12 @@ static void ucp_ep_delete(ucp_ep_h ep)
     ucs_callbackq_remove_if(&ep->worker->uct->progress_q,
                             ucp_wireup_msg_ack_cb_pred, ep);
     if (!(ep->flags & UCP_EP_FLAG_INTERNAL)) {
-        ucp_worker_keepalive_remove_ep(ep);
+        if (ep->cfg_index != UCP_WORKER_CFG_INDEX_NULL) {
```

Comment:
why needed?

Reply:
since we can remove EP from KA, when it is not added to KA

Reply:
ucp_worker_keepalive_remove_ep works even if ep was not added
for example, if err handling off

Reply:
ah I see, this doesn't work since I added an assertion there
removed the assertion and this extra `if`